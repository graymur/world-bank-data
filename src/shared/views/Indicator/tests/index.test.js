/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';
import {Indicator} from '../';
import _ from 'lodash';

const props = {
	match: {
		params: {indicatorId: '1'}
	},
	indicator: {
		id: '1',
		name: 'Indicator name',
		sourceNote: 'Source note',
		sourceOrganization: 'Source organization',
	},
	years: [2000, 2001],
	suggestData: {}
};

test('Renders indicator info', () => {
	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Indicator {...props}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Calls loadIndicator prop if indicator is not set', () => {
	const thisProps = _.omit(props, 'indicator');

	thisProps.loadIndicator = jest.fn();

	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Indicator {...thisProps}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	expect(thisProps.loadIndicator.mock.calls.length).toBe(1);
	expect(thisProps.loadIndicator.mock.calls[0][0]).toBe('1');
});
