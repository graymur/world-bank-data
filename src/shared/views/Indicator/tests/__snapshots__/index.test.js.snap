// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calls loadIndicator prop if indicator is not set 1`] = `
<div
  className="indicator"
/>
`;

exports[`Renders indicator info 1`] = `
<div
  className="indicator"
>
  <div
    className="indicator"
  >
    <h1>
      Indicator name
    </h1>
    <p>
      Source note
    </p>
    <p>
      Source: 
      Source organization
    </p>
    <div
      className="indicator__years"
    >
      <select
        className="styled-select"
        defaultValue=""
        id="year"
        onChange={[Function]}
      >
        <option
          hidden={true}
          value=""
        >
          Select year
        </option>
        <option
          data-year={2000}
        >
          2000
        </option>
        <option
          data-year={2001}
        >
          2001
        </option>
      </select>
    </div>
    <div
      className="indicator__data"
    />
  </div>
</div>
`;
